sdk/monitor/azure-monitor-opentelemetry-autoconfigure/src/main/java/com/azure/monitor/opentelemetry/autoconfigure/implementation/utils/VersionGenerator.java

```diff
@@ -25,8 +25,19 @@ public final class VersionGenerator {
         artifactName = properties.get("name");
         artifactVersion = properties.get("version");
 
-        sdkVersionString = "java" + getJavaVersion() + getJavaRuntime() + ":" + "otel" + getOpenTelemetryApiVersion()
-            + ":" + "ext" + artifactVersion;
+        sdkVersionString = getResourceProvider() + "java" + getJavaVersion() + getJavaRuntime() + ":" + "otel"
+            + getOpenTelemetryApiVersion() + ":" + "ext" + artifactVersion;
+    }
+
+    private static String getResourceProvider() {
+        if (!Strings.isNullOrEmpty(System.getenv("WEBSITE_SITE_NAME"))) {
```

Comment:
Are these lookups only something that'd be set with environment variables or could they also be set with JVM properties? If it could be both this is not covering all cases.

Reply:
These environment variables are automatically populated by App Service, Azure Spring Cloud or AKS.

The goal is to detect which Azure environment is being used